internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -67,7 +67,8 @@ var (
 )
 
 // convertSidecar converts the manifest sidecar configuration into a format parsable by the templates pkg.
-func convertSidecar(s map[string]*manifest.SidecarConfig) ([]*template.SidecarOpts, error) {
+func convertSidecar(s map[string]*manifest.SidecarConfig, lbmft *manifest.LoadBalancedWebService, bsmft *manifest.BackendService, sjmft *manifest.ScheduledJob, wsmft *manifest.WorkerService) ([]*template.SidecarOpts, error) {
```

Comment:
Instead of having a giant function to handle all type of manifest. Can we have one for each manifest type and within each function we can extract common logic to avoid duplication? Plus, I feel like, for example lbws, we just need `LoadBalancedWebServiceConfig` as the only input.

Reply:
I get your point, but the trouble here is we are getting `manifest.SidecarConfig` as an input to this function and we need parent struct of it which is `LoadBalancedWebServiceConfig` which you mentioned in the later part. So even if we create one for each manifest type then also inside sidecar we need to find out which manifest type this sidecar belongs to? For that we need manifest type as an input to the `convertSidecar` function.

Reply:
I think Im pretty sure the signature can be much more simpler than this. It can be either

```golang
func convertLBWSSidecar(config *manifest.LoadBalancedWebServiceConfig) ([]*template.SidecarOpts, error)
```

or a generic one

```golang
func convertSidecars(config map[string]*manifest.SidecarConfig, opts ...additionalSidecarOpts) ([]*template.SidecarOpts, error) {
  ......
  for _, opt := range opts {
     opt(sidecars)
  }
  return sidecars, nil
}

type additionalSidecarOpts func(opts []*template.SidecarOpts)

func withAdditionalLBPortMapping(routeingRules) additionalSidecarOpts

// and `withAdditionalSCPortMapping` stuff like that
```
and in additionalSidecarOpts we define the logic how we can

Reply:
If my intuition is correct, I think this function also get simplified with the refactor suggested by Wanxian and I above. The logic will be:

```go
var out []*template.PortMapping
for _, name := range sidecarNames {
   for _, port := range exposedPorts {
       if port.ContainerName != name {
           continue
       }
       out = append(out, convertExposedPort(port))
   }
}
```

So there won't be a need to distinguish between types because each manifest type will implement the `ExposedPorts` method.

Reply:
I have made changes as per the discussion here. Do let me know if my changes infers your expectations.